designer/deployment-manager-api/src/main/scala/pl/touk/nussknacker/engine/api/deployment/DeploymentManager.scala

```diff
@@ -77,6 +77,11 @@ trait DeploymentManager extends AutoCloseable {
     } yield stateWithFreshness
   }
 
+  /**
+    * We provide a special wrapper called WithDataFreshnessStatus to ensure that fetched data is restored
+    * from the cache or not. If you use any kind of cache in your DM implementation please wrap result data
+    * with WithDataFreshnessStatus(data, cached = true)
```

Comment:
... in opposite situation use `WithDataFreshnessStatus(data, cached = false)`
BTW if we make this `WithDataFreshnessStatus` more often used in our code, I think it's time to add two factory methods: `fresh(value)`, `cached(value)` and make the constructor / apply private. WDYT?